string.c

```diff
@@ -5477,13 +5482,16 @@ rb_pat_search(VALUE pat, VALUE str, long pos, int set_backref_str)
 
 /*
  *  call-seq:
- *     str.sub!(pattern, replacement)          -> str or nil
- *     str.sub!(pattern) {|match| block }      -> str or nil
+ *    sub!(pattern, replacement)   -> self or nil
+ *    sub!(pattern) {|match| ... } -> self or nil
+ *
+ *  Performs the specified substring replacement on +self+;
+ *  returns +self+ if any replacement occurred, +nil+ otherwise.
```

Comment:
This should probably mention it only does a single substitution, similar to how `sub` is documented.